src/test/java/healthy/lifestyle/backend/admin/users/service/UserAdminServiceTest.java

```diff
@@ -0,0 +1,188 @@
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.List;
+import java.util.Optional;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+import org.junit.jupiter.api.Assertions;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.Arguments;
+import org.junit.jupiter.params.provider.MethodSource;
+import org.mockito.InjectMocks;
+import org.mockito.Mock;
+import org.mockito.Spy;
+import org.mockito.junit.jupiter.MockitoExtension;
+import org.modelmapper.ModelMapper;
+
+@ExtendWith(MockitoExtension.class)
+class UserAdminServiceTest {
+
+    @InjectMocks
+    UserAdminServiceImpl userAdminService;
+
+    @Mock
+    private UserAdminRepository userAdminRepository;
+
+    @Mock
+    private CountryRepository countryRepository;
+
+    @Mock
+    private RoleRepository roleRepository;
+
+    @Spy
+    private TestUtil testUtil;
+
+    @Spy
+    ModelMapper modelMapper;
+
+    @ParameterizedTest
+    @MethodSource("multipleFilters")
+    void getUsersByFiltersTest_shouldReturnEntityResponseDtoList_whenIdAndParamsAreValid(
+            String roleName,
+            String username,
+            String email,
+            String fullName,
+            String countryName,
+            Integer age,
+            List<Integer> resultSeeds) {
+        Role roleUser = testUtil.createUserRole(1);
+        Role roleAdmin = testUtil.createAdminRole(2);
+        Country country1 = testUtil.createCountry(1);
+        Country country2 = testUtil.createCountry(2);
+
+        User user1 = testUtil.createUser(1, roleUser, country1);
+        User user2 = testUtil.createUser(2, roleUser, country2);
+        User user3 = testUtil.createUser(3, roleAdmin, country1);
+        User user4 = testUtil.createUser(4, roleAdmin, country2);
+
+        List<User> users = Arrays.asList(user1, user2, user3, user4);
+
+        users = users.stream()
+                .filter(user ->
+                        resultSeeds.contains(Integer.parseInt(user.getUsername().split("-")[1])))
+                .collect(Collectors.toList());
+
+        // Mocking repository method
+        when(countryRepository.findById(anyLong())).thenReturn(Optional.of(new Country()));
+        when(roleRepository.findById(anyLong())).thenReturn(Optional.of(new Role()));
+        when(userAdminRepository.findByFilters(
+                        any(Role.class), eq(username), eq(email), eq(fullName), any(Country.class), eq(age)))
+                .thenReturn(Optional.of(users));
+
+        // When
+        List<UserResponseDto> result = userAdminService.getUsersByFilters(
+                roleUser.getName().equals(roleName) ? roleUser.getId() : roleAdmin.getId(),
+                username,
+                email,
+                fullName,
+                country1.getName().equals(countryName) ? country1.getId() : country2.getId(),
+                age);
```

Comment:
So if roleName/countryName is null, then your logic returns roleAdmin.getId()/country2.getId(). As I understand lines 88 and 92 are similar to random `long anyRoleId` and random `long anyCoutntryId`?